### PR TITLE
Signup header banner with refund policy in plans step

### PR DESCRIPTION
**Pull request by chriskmnds**
Related to https://github.com/Automattic/martech/issues/2790

## Proposed Changes

- Adds a prominent refund policy mention in a header banner on the plans step in signup. This is the last/4th variation in the design mocks: 
- This is behind a feature flag (`onboarding/emphasize-refund-period-in-plans-step`) and only affects the main/default signup flow (so `/start/plans`)

#### Desktop
Screenshot 2024-04-23 at 11 23 29 AM

#### Mobile (<480px)
Screenshot 2024-04-23 at 11 47 44 AM

## Testing Instructions

- Go to `/start/plans?flags=onboarding/emphasize-refund-period-in-plans-step`
- Confirm the banner with refund policy is shown
- Confirm back button is clickable and working
- Confirm it is only visible in plans step (so `/start/domains` or checkout should not show the banner)
- With the banner visible, select a plan (paid) and confirm the banner disappears when loading bar is visible and before going to checkout
- Confirm not visible when feature flag is disabled (`/start/plans?flags=-onboarding/emphasize-refund-period-in-plans-step`)

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by chriskmnds** [comment]
@isatuncman-auto this is ready if you want to test and leave some feedback. There is no clear "go" yet on the PT, but I went ahead and explored to get a feel for it and see the actual string on the screen (if it fits or needs revision). Can you maybe confirm fundamentally if this looks ok (also the wrapping on mobile)? 

any follow-ups and cleanup can happen during the time we allocated for the PT, along with testing the actual experiment hook. so this is just giving it a head start

**Comment by isatuncman-auto** [comment]
> @isatuncman-auto this is ready if you want to test and leave some feedback. There is no clear "go" yet on the PT, but I went ahead and explored to get a feel for it and see the actual string on the screen (if it fits or needs revision). Can you maybe confirm fundamentally if this looks ok (also the wrapping on mobile)?
> 
> any follow-ups and cleanup can happen during the time we allocated for the PT, along with testing the actual experiment hook. so this is just giving it a head start

I think it seems fine to me as long as design team is okay with that (and they seem to be okay). Again, this is a small test and we will see if we see any meaningful change in user behavior based on this change.

**Review by southp — APPROVED** [review]
Works as expected :) 
I've spot-checked the following flows with the feature flag enabled : 

* /start/plans -> see the banner
* /start/onboarding-pm -> no banner
* /plans while logged-in -> no banner
* /setup/newsletter -> no banner

It also works correctly on mobile scrolling up/down. LGTM :)

File: client/signup/signup-header/index.tsx [hunk 1]
```diff
@@ -44,27 +43,41 @@ const SignupHeader = ( {
 	);
 	const showProgressBar = progressBar.flowName !== FREE_FLOW;
```
Comment (southp) [inline client/signup/signup-header/index.tsx:44]: Just a side note that the progress bar that used to be shown on Stepper flows is no longer there in production; not sure if it's intentional 🤔

Comment (chriskmnds) [reply client/signup/signup-header/index.tsx:44]: @southp Quite an on-point observation I think. 

Also indirectly leads me to another (maybe obvious) deviation between Stepper & Start -> the latter controlling the loading screen via `shouldShowLoadingScreen`, and it happens within the current step, while the former doing it separately in another step (`processing`). So I think we are good in the logic as the issue with the lingering banner shouldn't be an issue beyond Start flows.

Comment (chriskmnds) [reply client/signup/signup-header/index.tsx:44]: > So I think we are good in the logic as the issue with the lingering banner shouldn't be an issue beyond Start flows.

However we'd still need to pass `progressBar` through as it's currently an empty object in Stepper. and maybe worth renaming, or ditching altogether in favor of separate `flowName`/`stepName` props.

**Comment by chriskmnds** [comment]
I will go ahead and merge here and leave the experiment one for when discussion concludes: https://github.com/Automattic/wp-calypso/pull/89602#discussion_r1602715500

**Comment by vinimotaa** [comment]
@chriskmnds can you better elaborate on this please?

**Comment by chriskmnds** [comment]
> @chriskmnds can you better elaborate on this please?

I think I've seen a few designs where callouts appear in the header banner with a light/white background e.g. in A4A mocks here https://wordpress.com/read/blogs/199613069/posts/2108 or the refund period mentioned here pdvytD-y0-p2#comment-395. I'm referring to these, it should be easy to render same banner above with dark/light background

**Comment by vinimotaa** [comment]
@chriskmnds if the concern is to adopt/implement the blue bar, I'm all in using a white background on the top bar (with the light gray background.) I even [mentioned my concern](https://dotcomdesignp2.wordpress.com/2024/03/27/emphasizing-the-14-day-refund-period-on-plans-page/#comment-11148) with the blue color (which was a suggestion/request) from another folks.

**Pull request merged**